### PR TITLE
Closes #1613: Add space to footer menu items

### LINKED ISSUE #1613: No space between list items makes footer menu links harder to read
## Problem/Motivation
As we moved footer links from our current website into our QS2 site, we noticed they were harder to read in the context where a longer menu item breaks to two lines. Our current site has additional vertical spacing between the menu items which mitigates this issue. 

In our real content, showing our current and QS2 sites, footer sub area of the page: "Albert B. Weaver Science-Engineering Library" is one long menu item

albert-b-weaver-science-engineering-library-example

Part-fake example, with the global footer:

long-menu-items-example

### To Reproduce

1. Add or edit a long menu link that's part of the global footer or included in a menu block the footer sub area of the page 
5. View the list of menu links at multiple screen widths

## Proposed resolution
Add vertical space between links when they appear in the list in the footer. Example:
added-space

### PR DESCRIPTION
**Pull request by danahertzberg**
## Description
Add a space between footer list items

## Related issues
Closes #1613 

## How to test
https://b1f8212a-7d02-4d91-83c8-6d0013ee6896.probo.build/

1. Add menu items with long titles
2. Add menu to footer
3. Verify menu title wraps and has spacing between items

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: themes/custom/az_barrio/az_barrio.theme [hunk 1]
```diff
@@ -577,7 +577,7 @@ function az_barrio_preprocess_breadcrumb(&$variables) {
  *
  * Pass proper logo to the system branding block.
  *
- * Arizona Digital issue #1012.
+ * Arizona Digital issue #1012, #1613.
```
Comment (joeparsons) [inline themes/custom/az_barrio/az_barrio.theme:580]:
```suggestion
 * @link https://github.com/az-digital/az_quickstart/issues/1012
 * @link https://github.com/az-digital/az_quickstart/issues/1613 
```

File: themes/custom/az_barrio/az_barrio.theme [hunk 2]
```diff
@@ -577,7 +577,7 @@ function az_barrio_preprocess_breadcrumb(&$variables) {
...
  * Footer menu links styles, using default theme templates.
```
Comment (joeparsons) [inline themes/custom/az_barrio/az_barrio.theme:581]:
```suggestion
 * Footer menu links styles, using default theme templates.
 * @link https://github.com/az-digital/az_quickstart/issues/1012
 * @link https://github.com/az-digital/az_quickstart/issues/1613 
```

**Comment by bberndt-uaz** [comment]
I think I would prefer the [suggestion from Bob Liu in the related issue](https://github.com/az-digital/az_quickstart/issues/1613#issuecomment-1661009401):

- Add the mb-1 class (a margin-bottom of 4px) to the `<li>` elements.
- Reduce the line-height for footer sub menus from 1.5 to 1.25. To target those lists specifically, we could add a CSS rule such as:

```
#footer_sub .nav-utility {
  line-height: 1.25;
}
```

This would only introduce a visible change to footer menu links which wrap to the next line; otherwise the footer menus would look the same.

**Comment by danahertzberg** [comment]
@bberndt-uaz I saw that suggestion. The only reason I didn't move forward with it is because the line height is consistent with body copy as that's where it is set. I have no preference, just initially thought about keeping consistency.

**Review by joeparsons — DISMISSED** [review]

**Comment by danahertzberg** [comment]
@tkmccarthy, @liquid06, Bob Liu -- Your review on here would be very much appreciated.

- Original footer spacing: https://quickstart.arizona.edu/
- New footer spacing: https://2b71a4ad-55ad-4e58-8777-13d2c1580ae7.probo.build/

Most noticeable on "Colleges, Schools, Departments" link on desktop

**Review by bberndt-uaz — APPROVED** [review]
I still have a slight preference for the more compact look of the current footer links, but I also see that on desktop the spacing looks cramped once you introduce any multi-line links. Since links are less likely to span multiple lines in the mobile view, I'll make one more commit suggestion that would keep the current spacing on mobile. If you would rather keep the bottom margin uniform across all screen sizes, feel free to ignore it!

File: themes/custom/az_barrio/az_barrio.theme [hunk 3]
```diff
@@ -629,7 +630,7 @@ function az_barrio_preprocess_block(&$variables) {
           // Add classes to interior <li> elements and <a> elements.
           if (!empty($variables['elements']['content']['#items'])) {
             foreach ($variables['elements']['content']['#items'] as $key => $item) {
-              $item['attributes']->addClass('d-flex', 'px-0');
+              $item['attributes']->addClass('d-flex', 'px-0', 'mb-2');
```
Comment (bberndt-uaz) [inline themes/custom/az_barrio/az_barrio.theme:633]:
```suggestion
              $item['attributes']->addClass('d-flex', 'px-0', 'mb-1', 'mb-md-2');
```

**Comment by bearwithbob** [comment]
+1 to @bberndt-uaz's comment. I also prefer keeping the current spacing on mobile, but am also fine if we'd like to keep the spacing consistent across screen sizes.

**Comment by danahertzberg** [comment]
Ready to merge!

**Review by bberndt-uaz — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]

**Review by sanjanans — APPROVED** [review]
approved

**Comment by tkmccarthy** [comment]
Sorry I didn't weigh in sooner, but I think this a good solution to help distinguish the second line from a new menu item. Spacing on mobile looks good to me as well . As long as the desired item is easy to tap on mobile without accidentally clicking the one above or below it, I'm happy. Thanks!

**Pull request merged**